### PR TITLE
landlord-4 hw

### PR DESCRIPTION
**Pull request by ghost**
comfort: 3
completeness: 3

I followed the way of tun.r exercises but still listing apartments and viewing keep making errors.
Need to understand params, connections and syntax in controllers.

### CODE REVIEW

File: controllers/tenants.rb [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
 
 # show
 get '/apartments/:id/tenants/new' do
+  @apartment = Apartment.find(params[:id])
   erb :"tenants/new"
 end
+
+post '/apartments/:id/tenants' do
+  @apartment = Apartment.find(params[:id])
+  @apartment.tenants.create(params[:apartment])
```
Comment (jshawl) [inline controllers/tenants.rb:15]: Excellent!!

File: views/apartments/new.erb [hunk 2]
```diff
@@ -1,8 +1,10 @@
+<h3> Add New Apartments </h3>
+
 <form method="post" action="/apartments">
-  <input type="text" placeholder="Address">
-  <input type="text" placeholder="Monthly Rent">
-  <input type="text" placeholder="Square Feet">
-  <input type="text" placeholder="Number Beds">
-  <input type="text" placeholder="Number Baths">
+  <input type="text" placeholder="Address" name="apartment[address]">
```
Comment (jshawl) [inline views/apartments/new.erb:4]: excellent use of namespacing!

File: views/tenants/new.erb [hunk 3]
```diff
@@ -1 +1,8 @@
-form for adding a new tenant
+<h2>Add New Tenant </h2>
+
+<form method="post" action="/apartments/<%= @apartment.id %>/tenants">
+  <input name="apartment[:name]" type="text">
```
Comment (jshawl) [inline views/tenants/new.erb:4]: remove the colon from the brackets here

File: views/tenants/new.erb [hunk 4]
```diff
@@ -1 +1,8 @@
-form for adding a new tenant
+<h2>Add New Tenant </h2>
+
+<form method="post" action="/apartments/<%= @apartment.id %>/tenants">
```
Comment (jshawl) [inline views/tenants/new.erb:3]: nice job creating this action url!

**Comment by jshawl** [comment]
:+1: Nice job on this assignment! Which errors are you running into? Let me know if you'd like to review this exercise in our one on one this week.

**Comment by ghost** [comment]
Thank you!

I removed colons on brackets, and it seems everything okay to me but when I
clicked the listing and viewing apartments, error messages are shown. They
are about 'undefined id'.  I'd like to review this exercise in one on one!

**Pull request closed without merging**